### PR TITLE
Add spacing rules for jsx including mandatory space after self closing tag.

### PR DESCRIPTION
**Pull request by jcaianirh**
Enforce spacing rules according to: 

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-tag-spacing.md

Something Sam and I spoke about in a code review a while back. 

Enforces the following to be success:

```
<Hello />
<Hello
  firstName="John"
  lastName="Smith"
/>
```
Following to be error:

```
<Hello/>
<Hello firstName="John" lastName="Smith"/>
```

yarn run eslint --fix will automagically fix all spacing rules for you.

### CODE REVIEW

File: frontend/.eslintrc [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "react/no-string-refs": 1,
     "react/jsx-uses-react": "error",
     "react/jsx-uses-vars": "error",
+    "react/jsx-tag-spacing": 2,
```
Comment (benjaminapetersen) [inline frontend/.eslintrc]: I am a fan of rules that eliminate the need for us to `nit` at this stuff, +1!

**Comment by spadgett** [comment]
+1 to having this rule. Seems like the best way to enforce a code style. I like the approach of enabling these rules one at a time as well.

I might hold off until a couple PRs merge first though to avoid conflicts. Fixing up the spaces happens automatically with `--fix` right?

**Comment by spadgett** [comment]
It would be good to get changes like this in before we branch to avoid conflicts when cherry-picking fixes

**Comment by jcaianirh** [comment]
@spadgett Yes, using--fix will fix all spacing.

**Comment by spadgett** [comment]
OK please rebase, and I think we can go ahead with this one.

**Pull request merged**